blocks/bricks/recurrent.py

```diff
@@ -630,3 +631,231 @@ def apply(self, *args, **kwargs):
     @apply.delegate
     def apply_delegate(self):
         return self.children[0].apply
+
+
+class RecurrentStack(BaseRecurrent, Initializable):
+    u"""Stack of recurrent networks.
+
+    Build a stack of recurrent layers from a supplied list of
+    BaseRecurrent objects. Each object must have a `sequences`,
+    `contexts`, `states` and `outputs` parameters to its `apply` method.
+    It is assumed that all `states` have a "states" element
+    (this can be configured with `states_name` parameter.)
+
+    The `sequences` (the "inputs") of the the first layer (layer 0) is
+    used as the `sequence` of the stack (the instance of this class.)
+    The "mask" element of the `sequences` is carried to all layers that
+    have a "mask" element in their `sequences`.
+    For all the following layers, all elements of the `sequences`, except
+    for "mask", are forked from the "states" element of the of the
+    `states` of the previous layer.
+
+    The concatenated `states` of all layers is the `states` of the stack.
+    In order to avoid conflict, the names of the elements appearing in the
+    `states` of each layers are suffixed with a '_<n>' before combining
+    them. The '<n>' is the number of the layer starting from 0.
+    The same process is true for the `contexts` and `outputs` of the
+    stack.
+
+    Parameters
+    ----------
+    transitions : list
+        List of recurrent units to use in each layer.
+        Note: A suffix with layer number is added to transitions' names.
+    prototype : :class:`.FeedForward`, optional
+        The transformation applied to each states of each layer  when it
+        is forked to the sequences of the next layer. By default a
+         :class:`.Linear` transformation with biases is used.
+    states_name : string
+        In a stack of RNN the state of each layer is used as input to the
+        next. This string identify which element in the output of each
+        layer should be used for this task. By default the element is
+        called "states". To be more precise, this is the name of the
+        element in the outputs of the apply method of each transition
+        (layer) that is used, via fork, as the sequences (input) of the
+        next layer. The same element should also appear in the states
+        parameter of the apply method.
+    fast : bool
+        Use the fast, but also memory consuming, implementation of this
+        code. By default true.
+        WARNING: there is something strange in the
+        behaviour of the slow implementation:
+        Only outputs, used outside this class, can be found using the
```

Comment:
That's probably because they are in the inner graph, which by the way can be accessed using `scan_variables` attribute of `ComputationGraph`.